code/lib/core-server/src/utils/new-story-templates/typescript.ts

```diff
@@ -0,0 +1,33 @@
+import dedent from 'ts-dedent';
+
+export function getTypeScriptTemplateForNewStoryFile(data: {
+  /** The components file name without the extension */
+  basename: string;
+  componentExportName: string;
+  default: boolean;
+  /** The framework package name, e.g. @storybook/nextjs */
+  frameworkPackageName: string;
+  /** The exported name of the default story */
+  exportedStoryName: string;
+}) {
+  const importName = data.default ? 'Component' : data.componentExportName;
+  const importStatement = data.default
+    ? `import ${importName} from './${data.basename}'`
+    : `import { ${importName} } from './${data.basename}'`;
+
+  return dedent`
+  import type { Meta, StoryObj } from '${data.frameworkPackageName}';
+
+  ${importStatement};
+
+  const meta = {
+    component: ${importName}
+  } satisfies Meta<typeof ${importName}>
+
+  export default meta;
+
+  type Story = StoryObj<typeof meta>;
+
+  export const ${data.exportedStoryName}: Story = {}
```

Comment:
Missing a final semicolon here. Tbh we should probably respect their preferences here.

Reply:
Usually, we run prettier over the code, so that it matches the user's prettier configuration. Nowadays, the formatting landscape is much more fragmented, though (rome, eslint, prettier, biome, ...). Michael and I had already a discussion of how to provide code to the user, which is formatted by respecting their tooling and rules. As we have not yet decided on this, let alone implemented it, I would not do anything here for the time being